plugins/msg-broker/mcollective/lib/openshift/mcollective_application_container_proxy.rb

```diff
@@ -3311,9 +3311,11 @@ def self.select_best_fit_node_impl(server_infos)
         # Take all nodes with > the average remaining capacity.  Make sure to take at least max(4, 20% of nodes) nodes.
         server_infos.sort_by! { |server_info| server_info.node_consumed_capacity }.reverse!
         node_consumed_capacities = server_infos.map { |server_info| server_info.node_consumed_capacity }
-        average_consumed_capacity = (node_consumed_capacities.inject(0.0) { |sum, c| sum + c } / node_consumed_capacities.length) + 1
+        average_consumed_capacity = (node_consumed_capacities.inject(0.0) { |sum, c| sum + c } / node_consumed_capacities.length)
+        # Add a little so average isn't as harsh at the low end
+        cut_off_capacity = average_consumed_capacity + (1 - average_consumed_capacity/100)
```

Comment:
I may be missing the point here. `(1 - average_consumed_capacity/100)` will give:

* 1 when average is 0 - at which point everything is 0 so it doesn't matter which you keep or remove
* 0 or less when average is 100+ - at which point everything (or almost everything) is full so again it doesn't matter that you're removing (over-)full nodes.
* between 0 and 1 otherwise - tipping the cutoff point just above the average, which will only matter if it tips it past the next integer (I think capacity is stored as an integer). Is there really a use case where this makes a notable difference?

I don't have any particular objection other than it seems a bit more complicated/confusing than it needs to be.